### PR TITLE
[Vue | Frontend] First draft of Wolox Vue Style Guide.

### PR DESCRIPTION
**Pull request by BJordi**
**Remember** to use the [corresponding tag](../README.md#contributing) in this PR's title.

## Summary

First draft of Wolox Vue Style Guide. It includes:
- Basic rules
- Folder structure
- Ordering
- Props

## Screenshot

image

## Known issues / Notes

There's actually just one basic rule at the moment. Some more will be added in future PR's.

### CODE REVIEW

File: frontend/vue/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+|      ├ RelatedComponent1.vue
+|      └ RelatedComponent2.vue
+|
+├── config
+|    ├ api.js
+|    └ i18n.js
+|
+├── constants // Folder with js files
+|
+├── mixins // Folder with js files
```
Comment (mcallegari10) [inline frontend/vue/style-guide.md]: I'd add a mixin as an example, like `userType.js` or something like that

File: frontend/vue/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+|
+├── scss
+|    ├ application.scss
+|    ├ commons // Common styles, like display, margins, texts, etc.
```
Comment (mcallegari10) [inline frontend/vue/style-guide.md]:
```suggestion
|    ├ commons // SASS Component and common styles (display, margins, texts, etc.)
```

File: frontend/vue/style-guide.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+|    └ variables // Common SCSS variables, like colors, sizes, etc.
```
Comment (mcallegari10) [inline frontend/vue/style-guide.md]:
```suggestion
|    └ variables // SASS variables (colors, sizes, etc.)
```

File: frontend/vue/style-guide.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+## Props
+
+### Prop name casing
+
+As Vue separates the template from the script, we're going to follow the conventions of each language. HTML attribute names are case-insensitive, so browsers will interpret any uppercase characters as lowercase. That's why we're going to use kebab-case in the HTML. Within Javascript, lowerCamelCase is more natural.
+
+```pug
+// bad
+<template lang='pug'>
```
Comment (mcallegari10) [inline frontend/vue/style-guide.md]:
```suggestion
<template lang="pug">
```

File: frontend/vue/style-guide.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+<script>
+  props: {
+    'greeting-text': String
+  }
+</script>
+```
+
+```pug
+// good
+<template lang='pug'>
```
Comment (mcallegari10) [inline frontend/vue/style-guide.md]:
```suggestion
<template lang="pug">
```

File: frontend/vue/style-guide.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  my-component(greeting-text='hi')
+</template>
+
+<script>
+  props: {
+    greetingText: String
```
Comment (mcallegari10) [inline frontend/vue/style-guide.md]:
```suggestion
    greetingText: { type: String }
```

File: frontend/vue/style-guide.md (hunk 2 above)
Comment (BJordi) [reply frontend/vue/style-guide.md]: SASS or SCSS?

Comment (mcallegari10) [reply frontend/vue/style-guide.md]: Since the tool is SASS, I'd use SASS

**Pull request merged**